### PR TITLE
Improve dinov2 accuracy in classification tasks.

### LINKED ISSUE #4288: Improve DinoV2 performance for classification tasks
We have an existing implementation of DinoV2 (https://github.com/openvinotoolkit/training_extensions/blob/develop/src/otx/algo/classification/backbones/vision_transformer.py) in our repository, but it currently underperforms compared to vanilla CNNs in classification tasks, both in terms of accuracy and efficiency. This issue aims to investigate and improve the fine-tuning process of DinoV2, specifically focusing on fine-tuning only the linear classification head. However, full fine-tuning of the model can also be considered as a baseline for comparison.

Tasks:
1. Analyze Current Performance:

  -  Evaluate the accuracy and efficiency of the current DinoV2 implementation.
  -  Compare results with a "standard" CNN baseline (consider our implemented models in OTX like MobileNetV3, EfficientNetb0, EfficientNetV2-s, etc.).
2. Optimize Backbone and Linear Head Fine-Tuning: 

  - Identify potential improvements in the fine-tuning setup (e.g., learning rate, optimizer, data augmentation, loss function, input size and embedding dimension, etc.).
  - Implement changes while keeping them compatible with the current repository design.

 3. Consider Full Fine-Tuning (Baseline):

-  Experiment with full fine-tuning of DinoV2 as a baseline to compare performance gains.

 4. Benchmark and Report Results:
  -  Measure improvements in accuracy and efficiency. At least 4-5 diverse datasets should be considered with different number of images.
  -  Provide recipe (config) for e2e working solution (from torch fine-tuning to OpenVINO IR model)

### PR DESCRIPTION
**Pull request by gyuilLim**
### Summary

Add `init_values=1e-5` to dinov2-small to enable layer scaling.
Rename `reg_token` to `register_tokens` to correctly load pretrain weights.
(Since `reg_token` isn't used during fine-tuning, it doesn't affect performance)

Resolves https://github.com/open-edge-platform/training_extensions/issues/4288

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have ran e2e tests and there is no issues.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/open-edge-platform/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/open-edge-platform/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/open-edge-platform/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2025 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

## 🚀 Integration Run Summary

- **Last Commit:** Revert 'Fix LoRA Architecture'
- **Result:** ⚠️ SKIPPED
- **Workflow Run:** [View Run](https://github.com/open-edge-platform/training_extensions/actions/runs/15556932143)

### CODE REVIEW

**Comment by gyuilLim** [comment]
- EfficientNetv2

| DATASET         | Model          | Method            | VAL ACC (%) | TEST ACC (%) | Trainable Params |
|-----------------|----------------|--------------------|-------------|---------------|------------------|
| Flower photos   | EfficientNetv2 | Full Fine-Tuning   | 98.9        | 98.54         | 20.2M            |
| Leeds butterfly | EfficientNetv2 | Full Fine-Tuning   | 99.7        | 99.7          | 20.2M            |
| CUB-200         | EfficientNetv2 | Full Fine-Tuning   | 78.4        | 77.5          | 20.4M            |
| Flower photos   | EfficientNetv2 | Freeze Backbone    | 97.7        | 97.71         | 6.4K             |
| Leeds butterfly | EfficientNetv2 | Freeze Backbone    | 96.8        | 97.81         | 12.8K            |
| CUB-200         | EfficientNetv2 | Freeze Backbone    | 64.3        | 64.43         | 256K             |

- DINOv2(Previous)

| DATASET         | Model  | Method           | VAL ACC(%) | TEST ACC(%) | Trainable Params |
|-----------------|--------|------------------|------------|-------------|------------------|
| Flower photos   | DINOv2 | Full Fine-Tuning | 96.9       | 97.06       | 21.6M            |
| Leeds butterfly | DINOv2 | Full Fine-Tuning | 80.1       | 75.99       | 21.6M            |
| CUB-200         | DINOv2 | Full Fine-Tuning | 7.8        | 8.06        | 21.7M            |
| Flower photos   | DINOv2 | Freeze Backbone  | 52.2       | 52.17       | 1.9K             |
| Leeds butterfly | DINOv2 | Freeze Backbone  | 20.9       | 20.36       | 1.9K             |
| CUB-200         | DINOv2 | Freeze Backbone  | 2.3        | 1.89        | 77.0K            |

- DINOv2(`init_value=1e-5`)

| DATASET         | Model  | Method           | VAL ACC(%) | TEST ACC(%) | Trainable Params |
|-----------------|--------|------------------|------------|-------------|------------------|
| Flower photos   | DINOv2 | Full Fine-Tuning | 99.4       | 99.45       | 21.6M            |
| Leeds butterfly | DINOv2 | Full Fine-Tuning | 98.2       | 99.63       | 21.6M            |
| CUB-200         | DINOv2 | Full Fine-Tuning | 66.8       | 68.22       | 21.7M            |
| Flower photos   | DINOv2 | Freeze Backbone  | 89.8       | 89.80       | 1.9K             |
| Leeds butterfly | DINOv2 | Freeze Backbone  | 87.2       | 85.09       | 1.9K             |
| CUB-200         | DINOv2 | Freeze Backbone  | 46.7       | 45.18       | 77.0K            |

With `Feeze_backbone=True`, dinov2's trainable params is fewer than efficientnetv2, so it seems to performs lower.

**Review by sovrasov — APPROVED** [review]
Thanks for the PR and a detailed set of experiments!

**Review by kprokofi — APPROVED** [review]
